src/main/java/frc/trigon/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,95 @@
+package frc.trigon.robot.subsystems.elevator;
+
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.trigon.robot.subsystems.MotorSubsystem;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXMotor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXSignal;
+import org.trigon.utilities.Conversions;
+
+public class Elevator extends MotorSubsystem {
+    private final TalonFXMotor motor = ElevatorConstants.MASTER_MOTOR;
+    private final MotionMagicVoltage positionRequest = new MotionMagicVoltage(0).withEnableFOC(ElevatorConstants.FOC_ENABLED);
+    private final VoltageOut voltageRequest = new VoltageOut(0).withEnableFOC(ElevatorConstants.FOC_ENABLED);
+    private ElevatorConstants.ElevatorState targetState = ElevatorConstants.ElevatorState.REST;
+
+    public Elevator() {
+        setName("Elevator");
+    }
+
+    public boolean atTargetState() {
+        double targetStateOffsetDifference = Math.abs(targetState.targetPositionRotations - motor.getSignal(TalonFXSignal.POSITION));
+        return rotationsToMeters(targetStateOffsetDifference) < ElevatorConstants.TARGET_OFFSET_METERS;
+    }
```

Comment:
1. normal public should be under override functions
2. final
3. the variable can just be renamed to difference
4. test the tolerance in rotations